hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineDiffHelper.java

```diff
@@ -67,20 +67,58 @@ public static TimelineDiffResult getNewInstantsForIncrementalSync(HoodieTimeline
             + lostPendingCompactions);
         return TimelineDiffResult.UNSAFE_SYNC_RESULT;
       }
+
+      // Rather than checking for compaction instants with completed check and a commit file,
```

Comment:
trying to see if this is a mandatory fix. can we leave it as is. trying to not make any changes to core flow.